src/components/appBar/AppBar.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { StyleSheet, View, ViewStyle } from 'react-native';
+import { Appbar as BaseAppBar } from 'react-native-paper';
+import { useTheme } from '../../styles/themes';
+import { Headline } from '../typography/Headline';
+
+interface Props {
+    text: string;
+    onPress: () => void;
+    iconName?: string;
+    style?: ViewStyle;
+    children?: JSX.Element;
```

Comment:
Can we constraint the children type to the "Menu" component here ?

Like ?
```ts
children?: Menu;
```

It could be great to have an IDE warning that someone tries to include anything in the AppBar, like:

```html
<AppBar>
    <SomeBigImage/>
</AppBar>
```

Reply:
i tried to do it with `ReactElement<typeof Menu>`, but it did not work, it still allows to pass any type of jsx.element

Reply:
Jeez !
Well, forget about it then :/